### PR TITLE
[AB#116814] API for Downloading  blacklist email addresses

### PR DESCRIPTION
**Pull request by mvdr**
See also [nfield-source](https://github.com/NIPOSoftwareBV/nfield-source/pull/7850)

[AB#116814](https://dev.azure.com/niposoftware/15ce0e91-931d-4fbf-9169-8c3dde412b54/_workitems/edit/116814)

### CODE REVIEW

File: Library/Services/Implementation/NfieldBlacklistService.cs [hunk 1]
```diff
@@ -29,6 +29,15 @@ namespace Nfield.Services.Implementation
     /// </summary>
     internal class NfieldBlacklistService : INfieldBlacklistService, INfieldConnectionClientObject
     {
+        public Task<string> GetAsync()
+        {
+            var uri = BlacklistUrl();
+
+            return Client.GetAsync(uri)
+                .ContinueWith(task => task.Result.Content.ReadAsStringAsync().Result)
```
Comment (gkappen) [inline Library/Services/Implementation/NfieldBlacklistService.cs:37]: Isn't this potentially dangerous (using `Result` on an async method). This should be awaited I think

**Review by gkappen — COMMENTED** [review]
I have doubts

File: Library/Services/Implementation/NfieldBlacklistService.cs (hunk 1 above)
Comment (gkappen) [reply Library/Services/Implementation/NfieldBlacklistService.cs:37]: Why is this method not simply `async`?
``` c#
var result = await Client.GetAsync(uri);
var blacklist = await result.Content.ReadAsStringAsync();

return blacklist;
```
Or am I missing some vital bit of info?

Comment (neilboyd) [reply Library/Services/Implementation/NfieldBlacklistService.cs:37]: We also copy-pasted some old code in our PR #352, but then changed it to use `await`.

File: Library/Services/INfieldBlacklistService.cs [hunk 2]
```diff
@@ -32,5 +32,11 @@ public interface INfieldBlacklistService
         /// <exception cref="ArgumentException"></exception>
         /// <returns><see cref="BlacklistUploadStatus"/></returns>
         Task<BlacklistUploadStatus> PostAsync(string blacklist);
+
+        /// <summary>
+        /// Downloads the blacklist as a tab delimited string, that can be stored as csv file.
+        /// </summary>
+        /// <returns></returns>
```
Comment (neilboyd) [inline Library/Services/INfieldBlacklistService.cs:39]:
```suggestion
```

File: Tests/Services/NfieldBlacklistTests.cs [hunk 3]
```diff
@@ -42,6 +42,23 @@ public NfieldBlacklistTests()
             _target.InitializeNfieldConnection(mockedNfieldConnection.Object);
         }
 
+        #region GetAsync
+
+        [Fact]
+        public void TestGetAsync_SurveyHasSample_ReturnsSample()
```
Comment (neilboyd) [inline Tests/Services/NfieldBlacklistTests.cs:48]: this can also be async/await

**Review by gkappen — APPROVED** [review]

**Review by neilboyd — APPROVED** [review]

**Pull request merged**